pkg/queue/manager.go

```diff
@@ -48,17 +53,59 @@ type Manager struct {
 	clusterQueues map[string]ClusterQueue
 	localQueues   map[string]*LocalQueue
 
+	wm        sync.RWMutex
+	queue     workqueue.Interface
```

Comment:
I would suggest to move the queue to clusterqueue_controller, as this is where it is needed. Still, we will keep the snapshots map in manager, so there is also access from LocalQueue in the future.